app/uk/gov/hmrc/pillar2submissionapi/models/requests/IdentifierRequest.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2submissionapi.models.requests
+
+import play.api.mvc.{Request, WrappedRequest}
+import uk.gov.hmrc.auth.core.Enrolment
+
+case class IdentifierRequest[A](
+  request:            Request[A],
+  userId:             String,
+  groupId:            Option[String] = None,
+  enrolments:         Set[Enrolment] = Set.empty,
```

Comment:
we can probably do without the enrolments and strip out the pillar2Id straight away. If there is no pillar2 enrolement, we should reject

Reply:
So i've taken this as enrolments can be replaced with just a clientId? Is this correct?

Reply:
I think richard is suggesting as we don't make use of the enrolments anywhere further in this repo so we can strip out just the pillar2Id in a similar way to how we do it in pillar2-frontend in the enrolment identifier action [here](https://github.com/hmrc/pillar2-frontend/blob/main/app/controllers/actions/EnrolmentIdentifierAction.scala#L184) 

But rather than have a seperate action as there are valid journeys without a pillar2Id in pillar2-frontend we could roll this into this Initial identifier action and just reject if pillar2 is not on the enrolment.